en/install-openstack.md

```diff
@@ -0,0 +1,580 @@
+already installed).  We'll use this value as the `fsid` in the following
+`ceph-mon.yaml` configuration file.
+
+The second parameter is a `monitor-secret` for the configuration file. This is
+generated on the MAAS machine by first installing the `ceph-common` package and
+then by typing the following:
+
+```bash
+ceph-authtool /dev/stdout --name=mon. --gen-key
+```
+The output will be similar to the following:
+
+```no-highlight
+[mon.]
+        key = AQAARuRYD1p/AhAAKvtuJtim255+E1sBJNUkcg==
+```
+
+This is what the configuration file looks like with the required parameters:
+
+```yaml
+ceph-mon:
+  fsid: "a1ee9afe-194c-11e7-bf0f-53d6"
+  monitor-secret: AQAARuRYD1p/AhAAKvtuJtim255+E1sBJNUkcg==
+```
+
+!!! Note:
+    Double quotes are needed around the `fsid` value.
+
+Finally, deploy and scale the application as follows:
+
+```bash
+juju deploy --to lxd:1 --config ceph-mon.yaml ceph-mon
+juju add-unit --to lxd:2 ceph-mon
+juju add-unit --to lxd:3 ceph-mon
+```
+
+With these additional relations:
+
+```bash
+juju add-relation ceph-osd ceph-mon
+juju add-relation nova-compute ceph-mon
+juju add-relation glance ceph-mon
+```
+
+### [Cinder][charmcinder]
+
+For Cinder block storage, use the following `cinder.yaml` file:
+
+```yaml
+cinder:
+  glance-api-version: 2
+  block-device: None
+```
+
+And deploy with this:
+
+```bash
+juju deploy --to lxd:1 --config cinder.yaml cinder
+```
+
+Relations:
+
+```bash
+juju add-relation nova-cloud-controller cinder
+juju add-relation cinder mysql
+juju add-relation cinder keystone
+juju add-relation cinder rabbitmq-server
+juju add-relation cinder:image-service glance:image-service
+juju add-relation cinder ceph-mon
+```
+
+### [Swift proxy][charmswiftproxy]
+
+Swift also needs a unique identifier, best generated with the `uuid` command
+used previously. The output UUID is used for the `swift-hash` value in the
+`wift-proxy.yaml` configuration file:
+
+```yaml
+swift-proxy:
+  zone-assignment: auto
+  swift-hash: "a1ee9afe-194c-11e7-bf0f-53d662bc4339"
+```
+!!! Note:
+    Double quotes are needed around the `swift-hash` value.
+
+Use the following command to deploy:
+
+```bash
+juju deploy --to lxd:0 --config swift-proxy.yaml swift-proxy
+```
+
+These are its two relations:
+
+```bash
+juju add-relation swift-proxy swift-storage
+juju add-relation swift-proxy keystone
+```
+
+### [NTP][charmntp]
+
+The final component we need to deploy is a Network Time Protocol client, to
+keep everything in time. This is added with the following simple command:
+
+```bash
+juju deploy ntp
+```
+
+These last few `add-relation` commands finish all the connections we need to
+make:
+
+```bash
+juju add-relation neutron-gateway ntp
+juju add-relation nova-compute ntp
+juju add-relation ceph-osd ntp
+```
+
+All that's now left to do is wait on the output from `juju status` to show when
+everything is ready (everything turns green, if your terminal support colour).
+
+## Deploy OpenStack as a bundle
```

Comment:
Mainly because I think there is probably going to be extra stuff around it, like tweaking stuff